src/redprl/machine.fun

```diff
@@ -154,10 +148,23 @@ struct
           NOMINAL => false
           (* An observation of apartness is only stable if one of the compared dimensions is bound. *)
         | CUBICAL =>
-            if dimensionSafeToObserve syms r1 orelse dimensionSafeToObserve syms r2 then 
-              false 
-            else
-              raise Unstable
+            let
+              fun isBound syms r =
+                case r of
+                   P.VAR x => SymSet.member syms x
+                 | P.APP _ => false
+              fun isConstant r =
+                case r of
+                   P.VAR _ => false
+                 | P.APP P.DIM0 => true
+                 | P.APP P.DIM1 => true
+                 | P.APP _ => E.raiseError (E.INVALID_DIMENSION (TermPrinter.ppParam r))
+            in
+              if isBound syms r1 orelse isBound syms r2 orelse (isConstant r1 andalso isConstant r2) then
+                false
+              else
+                raise Unstable
+            end
```

Comment:
The previous code allowed `x=1` where `x` is not bound.

Reply:
@favonia Great catch!!